mlrun/api/db/sqldb/db.py

```diff
@@ -665,6 +674,204 @@ def delete_project(self, session, name: str):
     def list_projects(self, session, owner=None):
         return self._query(session, Project, owner=owner)
 
+    def get_feature_set(
+        self, session, project: str, name: str, tag: str = None, hash_key: str = None
+    ) -> schemas.FeatureSet:
+        query = self._query(session, FeatureSet, name=name, project=project)
+        computed_tag = tag or "latest"
+        tag_fs_uid = None
+        if not tag and hash_key:
+            uid = hash_key
+        else:
+            tag_fs_uid = self._resolve_tag_function_uid(
+                session, FeatureSet, project, name, computed_tag
+            )
+            if tag_fs_uid is None:
+                raise mlrun.errors.MLRunNotFoundError(
+                    f"Feature-set tag not found {project}/{name}:{tag}"
+                )
+            uid = tag_fs_uid
+        if uid:
+            query = query.filter(FeatureSet.uid == uid)
+        obj = query.one_or_none()
+        if obj:
+            db_fs = self._transform_feature_set_model_to_scheme(obj)
+
+            # If connected to a tag add it to metadata
+            if tag_fs_uid:
+                db_fs.metadata.tag = computed_tag
+            return db_fs
+        else:
+            raise mlrun.errors.MLRunNotFoundError(
+                f"Feature-set not found {project}/{name}"
+            )
+
+    # TODO - IMPLEMENT QUERY BY LABEL!!!!!
+    def list_feature_sets(
+        self,
+        session,
+        project: str,
+        name: str = None,
+        tag: str = None,
+        state: str = None,
+        entities: List[str] = None,
+        features: List[str] = None,
+        labels: List[str] = None,
+    ) -> List[schemas.FeatureSet]:
+        # Find object IDs by tag, project and fs-name (which is a like query)
+        tag = tag or "latest"
+        tag_query = self._query(session, FeatureSet.Tag, project=project, name=tag)
+        if name:
+            tag_query = tag_query.filter(FeatureSet.Tag.obj_name.ilike(f"%{name}%"))
+        obj_uids = {row.obj_id: row.name for row in tag_query}
```

Comment:
There is already a _resolve_class_tag_uids function, and I didn't use it exactly because of the reason you mention - it looks for a specific name, and doesn't do a like query. I don't know if that's a bug or not - I think the behaviour there is intentional? If it isn't, then I can modify it and use it instead.

Reply:
There is no `_resolve_class_tag_uids` - there is `_resolve_class_tag_uid` (note the `s`)
and what you explained is exactly the wanted difference between them, `_resolve_class_tag_uid` is for get (specific) operation, in which a name must match exactly, `_resolve_class_tag_uids` is for the list operation, in which we want the `like` behavior, the naming is probably not the best, if you have a better one, go for it.
but anyways, follow what I said in the first comment - rename + fix

Reply:
I have modified the behaviour of `_resolve_class_tag_uids` to add a like query on name. However, I'm not gonna use it because it only returns the uids of the objects - as you look at the mechanism I put in place for tag assignment, I also want to get the list of all tags for each uid, and I can do it efficiently from the same query. So, I will not use that method in my code. We can modify the `list_functions` code to reuse my suggestion and then maybe extract it to a standalone function, but I propose to do it in another PR.